### ISSUE TITLE
behave fails with pyproject.toml without any [tool] sections

### ISSUE THREAD

**Issue by con-fah** [behave/behave#1202]
The new dev release https://pypi.org/project/behave/1.2.7.dev6/ won't work in some of our projects.
The problem is, that these projects a pyproject.toml that don't have an `[tool.X]`-like sections. `behave` will right away just throw an error in such projects, as can be seen below. Please also note, that _any_ `[tool]` section will prevent behave from this happening.

```bash
(venv) foo@foo-station:/home/foo/behave_repro$ behave
USING RUNNER: behave.runner:Runner
ConfigError: No steps directory in '/home/foo/behave_repro/features'

(venv) foo@foo-station:/home/foo/behave_repro$ touch pyproject.toml
(venv) foo@foo-station:/home/foo/behave_repro$ behave
Traceback (most recent call last):
  File "/home/foo/behave_repro/venv/bin/behave", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/home/foo/behave_repro/venv/lib/python3.11/site-packages/behave/__main__.py", line 289, in main
    config = Configuration(args)
             ^^^^^^^^^^^^^^^^^^^
  File "/home/foo/behave_repro/venv/lib/python3.11/site-packages/behave/configuration.py", line 787, in __init__
    load_configuration(self.defaults, verbose=self.verbose)
  File "/home/foo/behave_repro/venv/lib/python3.11/site-packages/behave/configuration.py", line 682, in load_configuration
    defaults.update(read_configuration(filename, verbose))
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/foo/behave_repro/venv/lib/python3.11/site-packages/behave/configuration.py", line 660, in read_configuration
    parsed = parse_func(path)
             ^^^^^^^^^^^^^^^^
  File "/home/foo/behave_repro/venv/lib/python3.11/site-packages/behave/configuration.py", line 588, in read_toml_config
    config_tool = config["tool"]
                  ~~~~~~^^^^^^^^
KeyError: 'tool'

(venv) foo@foo-station:/home/foo/behave_repro$ echo "[tool.foo]" > pyproject.toml 
(venv) foo@foo-station:/home/foo/behave_repro$ behave
USING RUNNER: behave.runner:Runner
ConfigError: No steps directory in '/home/foo/behave_repro/features'
```

**Comment by bittner** [comment]
Sounds like an obvious wrong assumption. Good catch! :+1: 

Would you be ready to supply a PR to fix that?

I guess, we should most obviously ignore `pyproject.toml` in that case and continue reading the other configuration files that we currently accept.

**Comment by jenisys** [comment]
@con-fah 
QUESTION: 
* Do you use any other `behave` config-file, like: `behave.ini` (in the repo dir or your HOME directory) ?

NOTE: You seem to have actually two problems in your repo: 
* The one where the `pyproject.toml` has no `[tool]` section and 
* the other that you seem to have a `features/` without a `steps/` subdirectory (as shown in the second try where the `ConfigError` exception is raised and handled in `behave`).

**Comment by con-fah** [comment]
@jenisys sorry for the confusion, that was more like a minimal setup to reproduce the bug, that's why there aren't `features/` and `steps/` directories.

Our projects usually still configure all the tools in setup.cfg and we started using pyproject.toml for building the wheels after PEP 517. And as we didn't have the need to move all the config from setup.cfg to pyproject.toml, there is no `[tool]` section in the `pyproject.toml`. And at lastly, we just happen to do every configuration for `behave` via CLI.

**Comment by con-fah** [comment]
@bittner yes, I can do that.

**Comment by jenisys** [comment]
FIXED IN: behave 1.2.7.dev7
